clang/include/clang/AST/DeclCXX.h

```diff
@@ -2010,6 +2031,25 @@ class CXXDeductionGuideDecl : public FunctionDecl {
   /// this is an implicit deduction guide.
   CXXConstructorDecl *getCorrespondingConstructor() const { return Ctor; }
 
+  /// Get the deduction guide from which this deduction guide was generated,
+  /// if it was generated as part of alias template deduction or from an
+  /// inherited constructor.
+  CXXDeductionGuideDecl *getSourceDeductionGuide() const {
```

Comment:
```suggestion
  const CXXDeductionGuideDecl *getSourceDeductionGuide() const {
```
Would that break anything?

Reply:
Yeah, we should be returning a `const` pointer from a `const` member function; we can add a non-`const` member function overload to return a non-`const` pointer if we need one.

Reply:
Fixed